src/chi-vue/src/components/data-table-bulk-actions/DataTableBulkActions.tsx

```diff
@@ -0,0 +1,149 @@
+import { Component, Prop, Vue } from 'vue-property-decorator';
+import { findComponent } from '@/utils/utils';
+import {
+  BULK_ACTIONS_CLASSES,
+  BUTTON_CLASSES,
+  CHECKBOX_CLASSES,
+  CLOSE_CLASS,
+  DIVIDER_CLASSES,
+  FORM_CLASSES,
+  ICON_CLASS,
+  ICON_CLASSES,
+  UTILITY_CLASSES,
+} from '@/constants/classes';
+import DataTable from '../data-table/DataTable';
+
+@Component({})
+export default class DataTableBulkActions extends Vue {
+  @Prop() bulkActionConfig!: any;
+
+  showBulkActions?: boolean = true;
+
+  _selectedRows(e: any) {
+    this.$emit('chiShowSelectedOnly', e.srcElement.checked);
+  }
+
+  mounted() {
+    const dataTableComponent = findComponent(this, 'DataTable');
+
+    if (dataTableComponent) {
+      (dataTableComponent as DataTable)._bulkActionsComponent = this;
+    }
+  }
+
+  toggleCloseBulkActions() {
+    this.$props.bulkActionConfig.selectedRow = false;
```

Comment:
This is a prop mutation and is an antipattern in Vue.
If you need this boolean for some conditional rendering of certain elements, please, define a boolean property on the class with the initial value of the prop and toggle that property instead of the prop coming from the parent component.

Reply:
This is required to handle a specific scenario when we close bulk actions when few check boxes are already selected and to populate bulk actions component back upon selection of another checkbox.